tables/Tables/BaseTable.cc

```diff
@@ -779,16 +770,16 @@ BaseTable* BaseTable::select (rownr_t maxRow, rownr_t offset)
         maxRow = nrow() - offset;
     }
     if (offset == 0  &&  maxRow == nrow()) {
-        return this;
+      return getSharedPtr();    // pointer to itself
```

Comment:
This comment can better be placed above the definition of `getSharedPtr();` and not repeated here.

Reply:
Replaced getSharedPtr by shared_from_this